app/helpers/description_formatter_helper.rb

```diff
@@ -0,0 +1,32 @@
+module DescriptionFormatterHelper
+
+  def format_description(text, truncate: false)
+    # sanitize. should have been sanitized on input, but just to
+    # be safe.
+    text = DescriptionSanitizer.new.sanitize(text).html_safe
+
+    # truncate, make sure to tell it not to escape again, we already have.
+    # TODO: This is not a good solution for HTML text, may truncate in the middle
+    # of a tag, have to fix.
+    text = truncate(text, escape: false, length: 400, separator: /\s/)
+
+    # And convert line breaks to paragraphs
+    text = simple_format(text)
+
+    # a sufia helper to turn URLs into links.
+    text = iconify_auto_link(text)
+
+    text
+  end
+
+
+  def format_description_for_index(field)
+    if field.is_a? Hash
```

Comment:
No, it's not coming from Solr, it's some thing in Blacklight. I don't totally understand it either, just copied it from existing BL/sufia code.